src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -398,7 +398,9 @@ private function createSpecificPriceIfNeeded(
             $order->{Configuration::get('PS_TAX_ADDRESS_TYPE', null, null, $order->id_shop)}
         );
 
-        if (!(new Number((string) $amount->getTaxIncluded()))->equals(new Number((string) $initialProductPriceTaxIncl))) {
+        // Use the same rounding decimals used to call Product::getPriceStatic
+        $amountTaxIncluded = Tools::ps_round($amount->getTaxIncluded(), 2);
```

Comment:
We should use `PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\Precision::DEFAULT_PRECISION` wdyt?

Reply:
Seems good to me if I use it for Product::getPriceStatic too

Done ;)